app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -238,11 +238,48 @@ def refresh_cache
         end
       end
 
+      api :GET, "/compute_resources/:id/available_virtual_machines/", N_("List available virtual machines for a compute resource")
+      param :id, :identifier, :required => true
+      def available_virtual_machines
+        @available_virtual_machines = @compute_resource.vms.all
+      end
+
+      api :GET, "/compute_resources/:id/available_virtual_machines/:vm_id", N_("Show a virtual machine")
+      param :id, :identifier, :required => true
+      param :vm_id, :identifier, :required => true
+      def show_vm
```

Comment:
When I pass :vm_id that does not exists, the error message tells me the compute resource does not exist which is a bit misleading - the cr exists, the vm does not:

```
[vagrant@foreman-devel ~]$curl -H "Accept: application/json" -H "Content-Type: application/json" -k -X GET -u admin:changeme https://$(hostname)/api/v2/compute_resources/1/available_virtual_machines/1fe3645b-fc34-4830-850b-173fad6d8fa6 -d '{}'| json_reformat
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    77    0    75  100     2    295      7 --:--:-- --:--:-- --:--:--   296
{
    "error": {
        "message": "Resource compute_resource not found by id '1'"
    }
}

```

Reply:
I agree it should say that the VM is not found. I will fix it.